scipy/optimize/_basinhopping.py

```diff
@@ -248,12 +249,22 @@ class RandomDisplacement(object):
     Add a random displacement of maximum size, stepsize, to the coordinates
 
     update x inplace
+
+    Parameters
+    ----------
+    stepsize : float, optional
+        stepsize
+    rand_state : None or `np.random.RandomState` instance, optional
+        The random number generator that generates the displacements
     """
-    def __init__(self, stepsize=0.5):
+    def __init__(self, stepsize=0.5, rand_state=None):
         self.stepsize = stepsize
+        self.rand_state = rand_state
+        if rand_state is None:
+            self.rand_state = np.random.mtrand._rand
```

Comment:
Maybe use the utility `check_random_state`, https://github.com/scipy/scipy/blob/master/scipy/_lib/_util.py#L162
(which is also borrowed from scikit-learn)